Shell_Sort_and_Quick_Sort.cpp

```diff
@@ -0,0 +1,408 @@
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i <= walk ; i++ )
+							tempS[i] = temp[walk + head - i];
+					}	// if					
+				}	// if
+			}	// for			
+		}	// if
+		else if ( choice1 == 2 ) {	//  �u��ƨt�W 
+			for ( int walk = 0; walk < tempS.size(); walk++ ) {
+				if ( tempS[walk].dname != tempS[head].dname ) {					
+					QuickSortRecursive( tempS, head, walk - 1, choice2 );	// �ĤG���q�Ƨ� 
+					if ( show2 == '-' ) {									// �ƧǤ覡�� ����
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i < walk ; i++ )
+							tempS[i] = temp[walk + head - i - 1];
+					}	// if
+					
+					head = walk;
+				}	// if
+				
+				if ( walk == tempS.size() - 1 ) {	// �ƨ�̫���קK������X�j�� �S��̫�@�q�ƨ� 
+					QuickSortRecursive( tempS, head, walk, choice2 );	// �ĤG���q�Ƨ� 	
+					if ( show2 == '-' ) {								// �ƧǤ覡�� ����
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i <= walk ; i++ )
+							tempS[i] = temp[walk + head - i];
+					}	// if					
+				}	// if
+			}	// for	
+		}	// else if
+		else if ( choice1 == 3 ) {	//  �u��Ʀ~��
+			for ( int walk = 0; walk < tempS.size(); walk++ ) {
+				if ( tempS[walk].level != tempS[head].level ) {					
+					QuickSortRecursive( tempS, head, walk - 1, choice2 );	// �ĤG���q�Ƨ� 
+					if ( show2 == '-' ) {									// �ƧǤ覡�� ����
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i < walk ; i++ )
+							tempS[i] = temp[walk + head - i - 1];
+					}	// if
+					
+					head = walk;
+				}	// if
+				
+				if ( walk == tempS.size() - 1 ) {	// �ƨ�̫���קK������X�j�� �S��̫�@�q�ƨ� 
+					QuickSortRecursive( tempS, head, walk, choice2 );	// �ĤG���q�Ƨ� 	
+					if ( show2 == '-' ) {								// �ƧǤ覡�� ����
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i <= walk ; i++ )
+							tempS[i] = temp[walk + head - i];
+					}	// if					
+				}	// if
+			}	// for		
+		}	// else if
+		else if ( choice1 == 4 ) {	//  �u��ƤH�� 
+			for ( int walk = 0; walk < tempS.size(); walk++ ) {
+				if ( tempS[walk].total != tempS[head].total ) {					
+					QuickSortRecursive( tempS, head, walk - 1, choice2 );	// �ĤG���q�Ƨ� 	
+					if ( show2 == '-' ) {									// �ƧǤ覡�� ����
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i < walk ; )
+							tempS[i] = temp[walk + head - i - 1];
+					}	// if
+					
+					head = walk;
+				}	// if
+				
+				if ( walk == tempS.size() - 1 ) {	// �ƨ�̫���קK������X�j�� �S��̫�@�q�ƨ� 
+					QuickSortRecursive( tempS, head, walk, choice2 );	// �ĤG���q�Ƨ� 
+					if ( show2 == '-' ) {								// �ƧǤ覡�� ����
+						vector<collegeType> temp(tempS);
+						for ( int i = head; i <= walk ; i++ )
+							tempS[i] = temp[walk + head - i];
+					}	// if					
+				}	// if
+			}	// for	
+		}	// else if		
+	}	// else
+	
+ 	timer = clock() - timer; 	// get the elapse time
+ 	cout << endl << "[2] Quick Sort: " << timer << " clocks (";
+ 	cout << ((float)timer) / CLOCKS_PER_SEC << " seconds)." << endl;
```

Comment:
Duplicate logic
Quick sort time will be displayed by "WriteTimeRecords" function in line 335.